lib/modem_info/modem_info.c

```diff
@@ -447,18 +456,37 @@ int modem_info_string_get(enum modem_info info, char *buf)
 			  CONFIG_MODEM_INFO_BUFFER_SIZE,
 			  NULL);
 
-	/* modem_info does not yet support array objects, so here we handle
-	 * the supported bands independently as a string
-	 */
+	/* modem_info does not yet support array objects, so here we handle */
+	/* the supported bands independently as a string */
 	if (info == MODEM_INFO_SUP_BAND) {
 		strcpy(buf, recv_buf + sizeof("%XCBAND: ") - 1);
 		return strlen(buf);
 	}
+	if (info == MODEM_INFO_IP_ADDRESS) {
```

Comment:
I think the AT parser module could help in order to decode the AT event or response. Do you really need to do manual parsing?

Reply:
perhaps the at parser is where this should be handled?
the issue is that the response from the `+CGDCONT?` command is not handled correctly when there are multiple contexts.  also, what is the desired output format wrt ipv4 and ipv6 addresses?